glusterd2/commands/volumes/volume-start.go

```diff
@@ -98,14 +114,42 @@ func volumeStartHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	allNodes, err := peer.GetPeerIDs()
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	oldvolinfo := volinfo
+	//save volume information for transaction failure scenario
+	if err := txn.Ctx.Set("oldvolinfo", oldvolinfo); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
 	txn.Steps = []*transaction.Step{
 		{
 			DoFunc:   "vol-start.StartBricks",
 			UndoFunc: "vol-start.StartBricksUndo",
 			Nodes:    volinfo.Nodes(),
 		},
-	}
+		{
+			DoFunc:   "vol-start.XlatorActionDoStart",
+			UndoFunc: "vol-start.XlatorActionUndoStart",
+			Nodes:    volinfo.Nodes(),
+		},
 
+		{
+			DoFunc:   "vol-start.UpdateVolinfo",
```

Comment:
`glusterd2/commands/volumes/volume-option.go` has the order steps reversed but I think this should be okay as volfiles are regenrated anyway during volume set on stopped volume and daemons will get the new volfiles.